src/Lucene.Net/Util/NumericUtils.cs

```diff
@@ -149,10 +150,19 @@ private static int ThrowIllegalShift()
 		/// <param name="shift">how many bits to strip from the right
 		/// </param>
 		public static System.String LongToPrefixCoded(long val, int shift)
-		{
-			char[] buffer = new char[BUF_SIZE_LONG];
-			int len = LongToPrefixCoded(val, shift, buffer);
-			return new System.String(buffer, 0, len);
+        {
+            var pool = ArrayPool<char>.Shared;
```

Comment:
No need for this. Can use a thread static field instead. Less work to be done.

Reply:
will use stackalloc instead